game/src/sandbox/dashboards/trip_table.rs

```diff
@@ -214,216 +137,292 @@ fn make(ctx: &mut EventCtx, app: &App, opts: &Options) -> Panel {
+                    if app.primary.has_modified_trips {
+                        Checkbox::switch(
+                            ctx,
+                            "trips unmodified by experiment",
+                            None,
+                            state.unmodified_trips,
+                        )
+                    } else {
+                        Widget::nothing()
+                    },
+                    if app.primary.has_modified_trips {
+                        Checkbox::switch(
+                            ctx,
+                            "trips modified by experiment",
+                            None,
+                            state.modified_trips,
+                        )
+                    } else {
+                        Widget::nothing()
+                    },
+                    if any_congestion_caps {
+                        Checkbox::switch(
+                            ctx,
+                            "trips not affected by congestion caps",
+                            None,
+                            state.uncapped_trips,
+                        )
+                    } else {
+                        Widget::nothing()
+                    },
+                    if any_congestion_caps {
+                        Checkbox::switch(
+                            ctx,
+                            "trips affected by congestion caps",
+                            None,
+                            state.capped_trips,
+                        )
+                    } else {
+                        Widget::nothing()
+                    },
+                ]),
+            ])
+        }),
+        from_controls: Box::new(|panel| {
+            let mut modes = BTreeSet::new();
+            for m in TripMode::all() {
+                if panel.is_checked(m.ongoing_verb()) {
+                    modes.insert(m);
+                }
+            }
+            Filters {
+                modes,
+                off_map_starts: panel.is_checked("starting off-map"),
+                off_map_ends: panel.is_checked("ending off-map"),
+                unmodified_trips: panel
+                    .maybe_is_checked("trips unmodified by experiment")
+                    .unwrap_or(true),
+                modified_trips: panel
+                    .maybe_is_checked("trips modified by experiment")
+                    .unwrap_or(true),
+                uncapped_trips: panel
+                    .maybe_is_checked("trips not affected by congestion caps")
+                    .unwrap_or(true),
+                capped_trips: panel
+                    .maybe_is_checked("trips affected by congestion caps")
+                    .unwrap_or(true),
+            }
+        }),
+        apply: Box::new(|state, x| {
+            // TODO One big boolean expression?
+            let mut ok = true;
+            if !state.modes.contains(&x.mode) {
+                ok = false;
```

Comment:
could save some checks with early returns, but no biggie.

Reply:
Done! For some reason I expected `return` to not work from a closure. I think sometimes `?` in a closure has messed up for me.